CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -104,10 +148,25 @@ local function AddTool(tool)
 	UpdateLayout()
 end
 
+local humanoidChangedEvent = nil
+local function RegisterHumanoid(humanoid)
+	if humanoidChangedEvent then
+		humanoidChangedEvent:disconnect()
+		humanoidChangedEvent = nil
+	end
+	if humanoid then
+		humanoidChangedEvent = humanoid.Changed:connect(function() UpdateHealth(humanoid) end)
+		UpdateHealth(humanoid)
+	end
+end
+
 local function OnChildAdded(child)
 	if child:IsA("Tool") or child:IsA("HopperBin") then
 		AddTool(child)
 	end
+	if child:IsA("Humanoid") then
```

Comment:
Since `OnChildAdded` is used for backpack stuff too we might want to make sure child is in the character